### PR TITLE
chore(ci): fix usage of `a deprecated Node.js version`

### LINKED ISSUE #1230: `ci.yml` uses `a deprecated Node.js version`
[`ci.yml` uses `a deprecated Node.js version`](https://github.com/vercel/ncc/actions/runs/11851136072).

### PR DESCRIPTION
**Pull request by hamirmahal**
Fixes #1230

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,44 +23,44 @@ jobs:
...
-    - name: Maybe Release
-      if: matrix.os == 'ubuntu-latest' && matrix.node == 20 && github.event_name == 'push' && github.ref == 'refs/heads/main'
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        NPM_TOKEN: ${{ secrets.NPM_TOKEN_ELEVATED }}
-      run: npx semantic-release@19.0.5
+      - uses: actions/checkout@v4
+      # https://github.com/nodejs/node-gyp#installation
+      - name: Use Python 3.11
+        uses: actions/setup-python@v5
```
Comment (hamirmahal) [inline .github/workflows/ci.yml:29]: `v5` is the latest major version of this action at the time of this writing.

**Review by styfle — CHANGES_REQUESTED** [review]
Thanks for the PR!

Please revert all the whitespace changes so its clear what is changing. Changes to formatting can happen in a different PR than code changes.

**Comment by hamirmahal** [comment]
You're welcome!

I reverted the changes from formatting on save.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
     - uses: actions/checkout@v4
     # https://github.com/nodejs/node-gyp#installation
     - name: Use Python 3.11
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
```
Comment (styfle) [inline .github/workflows/ci.yml:29]:
```suggestion
      uses: actions/setup-python@v5
      -- force ci
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -26,7 +26,8 @@ jobs:
...
+      -- force ci
```
Comment (styfle) [inline .github/workflows/ci.yml:30]:
```suggestion
```

**Review by styfle — APPROVED** [review]

**Pull request merged**